### PR TITLE
Added StorageBaseTest and unified redis and jpa tests

### PR DESCRIPTION
**Pull request by quilombodigital**
I created a base storage test, with the tests that should be shared and performed at all storages.
I unified the redis and the jpa example to use it.
Some minimum small changes were made to other storage classes.

### CODE REVIEW

**Comment by JoeHegarty** [comment]
@DanielSperry can you take a lol at this as I am on the road at EARS.

**Comment by JoeHegarty** [comment]
Look rather

**Comment by DanielSperry** [comment]
I've LOL and liked ;). Merging it.
Thanks for your contribution, @quilombodigital.

File: actors/providers/redis/src/main/java/com/ea/orbit/actors/providers/redis/RedisStorageProvider.java [hunk 1]
```diff
@@ -60,13 +60,14 @@ public RedisStorageProvider() {
...
 				.withSetterVisibility(JsonAutoDetect.Visibility.NONE)
+				.withSetterVisibility(JsonAutoDetect.Visibility.NONE)
 				.withCreatorVisibility(JsonAutoDetect.Visibility.NONE));
 		redis = new Jedis(host, port);
 		return Task.done();
 	}
 
 	private String asKey(final ActorReference reference) {
-		String classname = ActorReference.getInterfaceClass(reference).getSimpleName();
+		String classname = ActorReference.getInterfaceClass(reference).getName();
```
Comment (johnou) [inline actors/providers/redis/src/main/java/com/ea/orbit/actors/providers/redis/RedisStorageProvider.java]: intentional?

File: actors/providers/redis/src/main/java/com/ea/orbit/actors/providers/redis/RedisStorageProvider.java [hunk 2]
```diff
@@ -60,13 +60,14 @@ public RedisStorageProvider() {
 				.withFieldVisibility(JsonAutoDetect.Visibility.ANY)
 				.withGetterVisibility(JsonAutoDetect.Visibility.NONE)
 				.withSetterVisibility(JsonAutoDetect.Visibility.NONE)
+				.withSetterVisibility(JsonAutoDetect.Visibility.NONE)
```
Comment (johnou) [inline actors/providers/redis/src/main/java/com/ea/orbit/actors/providers/redis/RedisStorageProvider.java]: ops? :)

**Pull request merged**